### DISCUSSION TITLE
使用npm为页面添加https后部分功能出现问题

### DISCUSSION THREAD

**Discussion by EnergeticMrMask** [MetaCubeX/metacubexd#1096] [Q&A] [answered]
我用nginxproxymanager为metacubexd和mihomo external-controller添加了代理并配置了https。~~因为看着浏览器提示不安全很碍眼~~
<img width="683" alt="image" src="https://github.com/user-attachments/assets/c1b5c9f8-bfc3-4684-8538-d5e7f2d5b787">
这个操作我在自己的其他服务上重复过很多次了，没翻过车。~~所以应该不是我的问题~~
而且所有流程都只涉及内网访问
然后这样添加，一起都看起来很美好
<img width="560" alt="image" src="https://github.com/user-attachments/assets/c1733ef6-efc2-4b2f-873d-ad8d4ffca67c">
只是有些功能不太正常
![image](https://github.com/user-attachments/assets/f42b9688-a9aa-4fca-aa99-d4188d1e57e6)
比如概览这里永远不会变。
![image](https://github.com/user-attachments/assets/43d2fe96-53d8-4e77-97e8-e7c600dc53f7)
![image](https://github.com/user-attachments/assets/8e6fb368-a896-41a5-8bc2-fef27e282495)
连接情况和日志也是空的。
但代理里是可以操作的，虽然一直显示0B/s。
![image](https://github.com/user-attachments/assets/a747acdd-5870-4bd9-a37a-be650abe7181)
设置似乎也没什么问题。
好像涉及“动态”的部分都不正常。
如果我使用api(ip:port)+ui(http)或者api(ip:port)+ui(ip:port)的方式则都是正常的。
如果用api(ip:port)+ui(https)的方式则在添加后端的时候会显示failed to fetch
看起来就是https导致的问题。
配置文件如下:
```
mixed-port: 7890
allow-lan: true
unified-delay: true
mode: global
log-level: info
external-controller: 0.0.0.0:9090
secret: "xxxx"
```
是不是还需要添加别的配置？文档里相关的内容留白太多了。
希望大佬们教教我

**Comment by Plutonium141** [comment, marked as answer]
动态部分涉及websocket，检查一下nginx的websocket相关的设置
> 如果用api(ip:port)+ui(https)的方式则在添加后端的时候会显示failed to fetch

浏览器不允许在一个https网页中发起http连接，此时面板会显示failed to fetch

**Reply by EnergeticMrMask** [reply]
感谢大佬，将mihomo本身的反代开启websocket后正常了，之前只试了metacubexd开启websocket，没想到要开mihomo本身的。
